packages/sources/blocksize-capital-test/src/endpoint/price.ts

```diff
@@ -0,0 +1,138 @@
+import { SingleNumberResultResponse, makeLogger } from '@chainlink/external-adapter-framework/util'
+import { WebsocketReverseMappingTransport } from '@chainlink/external-adapter-framework/transports/websocket'
+import {
+  PriceEndpoint,
+  PriceEndpointInputParameters,
+} from '@chainlink/external-adapter-framework/adapter'
+import { InputParameters } from '@chainlink/external-adapter-framework/validation'
+import { customSettings } from '../config'
+import { TransportGenerics } from '@chainlink/external-adapter-framework/transports'
+
+const logger = makeLogger('BlocksizeCapitalWebsocketEndpoint')
+interface BaseMessage {
+  jsonrpc: string
+  id?: string | number | null
+}
+
+export interface Message extends BaseMessage {
+  method: 'vwap'
+  params: {
+    updates: {
+      ticker: string
+      price?: number
+      size?: number
+      volume?: number
+    }[]
+  }
+}
+
+const inputParameters: InputParameters & PriceEndpointInputParameters = {
+  base: {
+    aliases: ['from', 'coin'],
+    type: 'string',
+    description: 'The symbol of symbols of the currency to query',
+    required: false,
+  },
+  quote: {
+    aliases: ['to', 'market'],
+    type: 'string',
+    description: 'The symbol of the currency to convert to',
+    required: false,
+  },
+}
+
+export type EndpointTypes = {
+  Request: {
+    Params: { base: string; quote: string }
```

Comment:
We can use the params we export from the framework for price endpoints here
```suggestion
    Params: PriceEndpointParams
```